Userland/Libraries/LibPthread/semaphore.cpp

```diff
@@ -36,25 +36,53 @@ int sem_close(sem_t*)
 
 int sem_destroy(sem_t* sem)
 {
-    pthread_mutex_destroy(&sem->mtx);
-    pthread_cond_destroy(&sem->cv);
+    errno = pthread_mutex_destroy(&sem->mtx);
+    if (errno != 0)
+        return -1;
+
+    errno = pthread_cond_destroy(&sem->cv);
+    if (errno != 0)
+        return -1;
+
     return 0;
 }
 
-int sem_getvalue(sem_t*, int*)
+int sem_getvalue(sem_t* sem, int* sval)
 {
-    VERIFY_NOT_REACHED();
+    errno = pthread_mutex_trylock(&sem->mtx);
+
+    if (errno == EBUSY) {
+        *sval = 0;
+        return 0;
+    }
+
+    if (errno != 0)
+        return -1;
+
+    *sval = sem->value;
+
+    pthread_mutex_unlock(&sem->mtx);
+
+    return 0;
 }
 
 int sem_init(sem_t* sem, int shared, unsigned int value)
 {
     if (shared)
         return ENOSYS;
 
-    if (pthread_mutex_init(&sem->mtx, nullptr) != 0)
+    if (value > SEM_VALUE_MAX)
+        return EINVAL;
+
+    if (value > SEM_VALUE_MAX)
```

Comment:
Dr. Posix says to return 0, or -1, not an errno value. Pre-existing though.
https://www.man7.org/linux/man-pages/man3/sem_init.3p.html

Reply:
Yup, good catch. I've updated the PR. :+1: